pimd/pim_upstream.c

```diff
@@ -745,6 +745,16 @@ void pim_upstream_switch(struct pim_instance *pim, struct pim_upstream *up,
 		bool new_use_rpt;
 		bool send_xg_jp = false;
 
+		/*
+		 * In FHR pimreg interface is needed all the time
+		 * inorder to send register packets.
+		 */
+		if (PIM_UPSTREAM_FLAG_TEST_FHR(up->flags) &&
+		    up->reg_state == PIM_REG_NOINFO &&
```

Comment:
so — this is conditional on `reg_state == PIM_REG_NOINFO`. How is this updated when that state changes? Is `pim_upstream_switch` being called on a change on that? Or is there some other code doing that already and this is just a missing situation where we forgot to check that?

Reply:
In this context, the reg_state will be updated when the upstream KAT expires.